Common/Source/Dialogs/dlgFlarmIGCDownload.cpp

```diff
@@ -0,0 +1,897 @@
+         return error;
+       }
+       pBlock[i] = inchar;
+       CRC_calc = crc_update (CRC_calc,pBlock[i]); if(error)  return error;
+   //    if(deb_) { StartupStore(TEXT("Block[%u]  %02X" ),i, pBlock[i]);}
+    }
+*blocksize -= 8;
+ if(CRC_calc != CRC_in)
+ {
+   {StartupStore(TEXT("Rec Block CRC error!" ));}
+    error = REC_CRC_ERROR;
+ }
+ else
+ {
+  if(error)
+    {StartupStore(TEXT("Rec Block error  received!" ));}
+  else
+    {if(deb_) StartupStore(TEXT("Rec Block received OK!" ));}
+ }
+  return error;
+}
+
+
+
+
+
+
+static void UpdateList(void) {
+if(wIGCSelectListList != NULL)
+{
+  wIGCSelectListList->ResetList();
+  wIGCSelectListList->Redraw();
+}
+}
+
+static void OnUpClicked(WndButton* Sender) {
+    if (IGC_Index > 0) {
+        IGC_Index--;
+    } else {
+        LKASSERT(iNoIGCFiles>0);
+        IGC_Index = (iNoIGCFiles - 1);
+    }
+    if(wIGCSelectListList != NULL)
+    {
+      wIGCSelectListList->SetItemIndexPos(IGC_Index);
+      wIGCSelectListList->Redraw();
+      if(wIGCSelectListListEntry)
+        wIGCSelectListListEntry->SetFocus();
+    }
+}
+
+static void OnDownClicked(WndButton* pWnd) {
+
+    (void)pWnd;
+
+    if (IGC_Index < (iNoIGCFiles - 1)) {
+        IGC_Index++;
+    } else {
+        IGC_Index = 0;
+    }
+    if(wIGCSelectListList != NULL)
+    {
+      wIGCSelectListList->SetItemIndexPos(IGC_Index);
+      wIGCSelectListList->Redraw();
+      if(wIGCSelectListListEntry)
+        wIGCSelectListListEntry->SetFocus();
+    }
+}
+
+
+
+
+static void OnMultiSelectListListInfo(WindowControl * Sender, WndListFrame::ListInfo_t *ListInfo) {
+
+    (void) Sender;
+
+    if (ListInfo->DrawIndex == -1) {
+        ListInfo->ItemCount = iNoIGCFiles;
+
+    } else {
+        IGC_DrawListIndex = ListInfo->DrawIndex + ListInfo->ScrollIndex;
+        IGC_Index = ListInfo->ItemIndex + ListInfo->ScrollIndex;
+    }
+
+}
+
+
+
+static void OnEnterClicked(WndButton* pWnd) {
+TCHAR Tmp[MAX_PATH ];
+
+    (void)pWnd;
+
+    if (IGC_Index >= iNoIGCFiles) {
+        IGC_Index = iNoIGCFiles - 1;
+    }
+    if(szIGCStrings[IGC_Index ] == NULL) return;
+
+      _stprintf(Tmp, _T("%s %s ?"),MsgToken(2404),szIGCStrings[IGC_Index ]);
+     if (MessageBoxX(Tmp, MsgToken(2404),  mbYesNo) == IdYes)  // _@2404 "Download"
+     {
+        /** check if file already exist and is not empty ************/
+		_sntprintf(Tmp, 200, _T("%s"), szIGCStrings[IGC_Index ]);
+		TCHAR* remaining;
+		TCHAR* Filename  = _tcstok_r(Tmp ,TEXT(" "),&remaining);
+
+		LocalPath(IGCFilename, _T(LKD_LOGS), Filename);
+		FILE* f = _tfopen(IGCFilename, TEXT("r")) ;
+		if(  f  != NULL)       // file exists
+		{
+		  fseek(f,0,SEEK_END);  // check file not empty
+		  int size = ftell(f);
+		  fclose(f); f = NULL;
+		  if(size > 10)         // is larger than 10 bytes (not empty)
+		    if (MessageBoxX(MsgToken(2416), MsgToken(2398), mbYesNo) == IdNo) // _@M2416_ "File already exits\n download anyway?"
```

Comment:
why so complicated ? file exist or not exists no need to check file size ....
right waty to check if file exist is  : "lk::filesystem::exist(filepath)"

Reply:
Wasn't aware we have this function. Thanks for the hint!